tekton/utils/pipeline_ref.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+
+// PipelineRef represents a reference to a Pipeline using a resolver.
+type PipelineRef struct {
+	// Resolver is name of a resolver to be used (e.g. git)
+	Resolver string `json:"resolver"`
+
+	// Params is a slice of parameters for a given resolver
+	Params []Param `json:"params"`
+}
+
+// Param defines the parameters for a given resolver in PipelineRef
+type Param struct {
+	// Name is the name of the parameter
+	Name string `json:"name"`
+
+	// Value is the value of the parameter
+	Value string `json:"value"`
+}
+
+// ToTektonPipelineRef converts a PipelineRef object to Tekton's own PipelineRef type and returns it.
+func (pr PipelineRef) ToTektonPipelineRef() tektonv1.PipelineRef {
+	params := tektonv1.Params{}
+
+	for _, p := range pr.Params {
+		params = append(params, tektonv1.Param{
+			Name: p.Name,
+			Value: tektonv1.ParamValue{
+				Type:      tektonv1.ParamTypeString,
+				StringVal: p.Value,
+			},
+		})
+	}
+
+	tektonPipelineRef := tektonv1.PipelineRef{
+		ResolverRef: tektonv1.ResolverRef{
+			Resolver: tektonv1.ResolverName(pr.Resolver),
+			Params:   params,
+		},
+	}
+
+	return tektonPipelineRef
+}
+
+// IsClusterScoped returns whether the PipelineRef uses a cluster resolver or not.
```

Comment:
I am a little confused why we have this function (and none for bundle/git)

Reply:
We only need this one so far for the debug mode. Can be added later if needed.

Reply:
Local/cluster scoped pipelines will only be allowed in debug mode. So this function will be used in that logic - if we're not in debug mode and it is cluster scope, we reject it.